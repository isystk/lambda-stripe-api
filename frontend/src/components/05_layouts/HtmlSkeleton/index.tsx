import React, { Children, FC, useState } from 'react'
import { connect } from '@/components/hoc'
import { ContainerProps, WithChildren } from 'types'
import useAppRoot from '@/stores/useAppRoot'
import Title, { TitleProps } from './Title'
import Head from 'next/head'
import { isReactElement } from '@/utils/general/object'
import { APP_NAME, APP_DESCRIPTION } from '@/constants'

/** HtmlSkeleton Props */
export type HtmlSkeletonProps = WithChildren & {
  showSideMenu?: boolean
}
/** Presenter Props */
export type PresenterProps = HtmlSkeletonProps & {
  title?: TitleProps['children']
  description: string
}

/** Presenter Component */
const HtmlSkeletonPresenter: FC<PresenterProps> = ({
  showSideMenu,
  main,
  title,
  description,
  children,
}) => (
  <>
    <Head>
      {/* タイトル */}
      <title>{title}</title>
      {/* favicon */}
      <link rel="icon" href="/favicon.ico" />
      {/* PCやモバイル（スマホ、タブレット）などのデバイスごとのコンテンツの表示領域 */}
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      {/* ブラウザテーマカラー */}
      <meta name="theme-color" content="#ffffff" />
      {/* サイト概要 */}
      <meta name="description" content={description} />
      {/* OGP 画像URL */}
      <meta property="og:image" content={'/ogp-image.png'} />
      {/* OGP タイトル */}
      <meta name="og:title" content={title} />
      {/* OGP サイト概要 */}
      <meta name="og:description" content={description} />
      {/* OGP Twitterカード */}
      <meta name="twitter:card" content="summary" />
      {/* apple ポータブル端末 アイコン */}
      <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
      {/* manifest.json */}
      <link rel="manifest" href="/manifest.json" />
      {/* Tailwind UI で採用されているカスタムフォント */}
      <link rel="stylesheet" href="https://rsms.me/inter/inter.css" />
    </Head>
    <Context.Provider value={main}>{children}</Context.Provider>
  </>
)

/** Container Component */
const HtmlSkeletonContainer: React.FC<
  ContainerProps<HtmlSkeletonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  let title: TitleProps['children'] | undefined = undefined
  const main = useAppRoot()

  if (!main) return <></>

  children = Children.map(children, (child) =>
    isReactElement(child) && child.type === Title
      ? (title = `${child.props.children} | ${APP_NAME}`) && undefined
      : child
  )
  if (!title) title = APP_NAME

  const description = APP_DESCRIPTION
  return presenter({
    main,
    title,
    description,
    children,
    ...props,
  })
}

export const Context = React.createContext(null)

export default connect<HtmlSkeletonProps, PresenterProps>(
  'HtmlSkeleton',
  HtmlSkeletonPresenter,
  HtmlSkeletonContainer
)

// Sub Component
export type { TitleProps }
export { Title }
