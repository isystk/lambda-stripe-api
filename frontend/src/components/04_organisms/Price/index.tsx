import React, { FC, useContext } from 'react'
import { ContainerProps, WithChildren } from 'types'
import * as styles from './styles'
import { connect } from '@/components/hoc'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import LineButton from '@/components/01_atoms/LineButton'
import ScrollIn from '@/components/01_atoms/ScrollIn'

/** PriceProps Props */
export type PriceProps = WithChildren
/** Presenter Props */
export type PresenterProps = PriceProps & {
  main
}

/** Presenter Component */
const PricePresenter: FC<PresenterProps> = ({ main, ...props }) => (
  <>
    <section className={`${styles.price}`}>
      <div className="flex items-center bg-black justify-center w-full">
        <div className="py-32 w-full">
          <ScrollIn>
            <p className="text-white font-bold text-3xl md:mb-16 text-center">
              PRICE
            </p>
          </ScrollIn>
          <div className="flex flex-wrap items-center p-8 md:mb-16">
            <div className="flex w-full md:w-1/2 justify-end items-center">
              <ScrollIn className="w-full md:w-72 md:mr-8">
                <div className="bg-white bg-opacity-20 rounded-lg py-16 h-64 mb-8 md:mb-0">
                  <p className="text-white text-center mb-8">
                    チャット回数制限あり
                  </p>
                  <p className="text-white font-bold text-center text-5xl">
                    無料
                  </p>
                </div>
              </ScrollIn>
            </div>
            <div className="flex w-full md:w-1/2 justify-start items-center">
              <ScrollIn className="w-full md:w-72 md:ml-8">
                <div className="bg-white bg-opacity-20 rounded-lg py-16 h-64 ">
                  <p className="text-white text-center mb-8">チャット無制限</p>
                  <div className="px-12">
                    <p className="text-white font-bold text-left text-4xl">
                      <span className="text-white text-xl mr-4">月額</span>¥980
                    </p>
                    <p className="text-white font-bold text-left text-4xl">
                      <span className="text-white text-xl mr-4">年額</span>
                      ¥9,800
                    </p>
                  </div>
                </div>
              </ScrollIn>
            </div>
          </div>
          <div className="flex items-center justify-center px-8">
            <LineButton link="#" label="友達に追加して質問してみる" />
          </div>
        </div>
      </div>
    </section>
  </>
)

/** Container Component */
const PriceContainer: React.FC<ContainerProps<PriceProps, PresenterProps>> = ({
  presenter,
  children,
  ...props
}) => {
  const main = useContext<MainService | null>(Context)
  if (!main) return <></>

  return presenter({
    children,
    main,
    ...props,
  })
}

export default connect<PriceProps, PresenterProps>(
  'Price',
  PricePresenter,
  PriceContainer
)
