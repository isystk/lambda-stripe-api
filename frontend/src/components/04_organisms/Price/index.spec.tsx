import React from 'react'
import renderer from 'react-test-renderer'
import Price from './index'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import '@testing-library/jest-dom/extend-expect'
import MainService from '@/services/main'

describe('Price', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))

    const component = renderer.create(
      <Context.Provider value={main}>
        <Price />
      </Context.Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
