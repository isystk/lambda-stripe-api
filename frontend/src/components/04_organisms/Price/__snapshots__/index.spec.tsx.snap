// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Match Snapshot 1`] = `
<section
  className="css-0"
>
  <div
    className="flex items-center bg-black justify-center w-full"
  >
    <div
      className="py-32 w-full"
    >
      <div
        className="undefined css-pjt9eq"
      >
        <p
          className="text-white font-bold text-3xl md:mb-16 text-center"
        >
          PRICE
        </p>
      </div>
      <div
        className="flex flex-wrap items-center p-8 md:mb-16"
      >
        <div
          className="flex w-full md:w-1/2 justify-end items-center"
        >
          <div
            className="w-full md:w-72 md:mr-8 css-pjt9eq"
          >
            <div
              className="bg-white bg-opacity-20 rounded-lg py-16 h-64 mb-8 md:mb-0"
            >
              <p
                className="text-white text-center mb-8"
              >
                チャット回数制限あり
              </p>
              <p
                className="text-white font-bold text-center text-5xl"
              >
                無料
              </p>
            </div>
          </div>
        </div>
        <div
          className="flex w-full md:w-1/2 justify-start items-center"
        >
          <div
            className="w-full md:w-72 md:ml-8 css-pjt9eq"
          >
            <div
              className="bg-white bg-opacity-20 rounded-lg py-16 h-64 "
            >
              <p
                className="text-white text-center mb-8"
              >
                チャット無制限
              </p>
              <div
                className="px-12"
              >
                <p
                  className="text-white font-bold text-left text-4xl"
                >
                  <span
                    className="text-white text-xl mr-4"
                  >
                    月額
                  </span>
                  ¥980
                </p>
                <p
                  className="text-white font-bold text-left text-4xl"
                >
                  <span
                    className="text-white text-xl mr-4"
                  >
                    年額
                  </span>
                  ¥9,800
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="flex items-center justify-center px-8"
      >
        <a
          className="css-0 p-2 pl-4 pr-4 flex items-center justify-center bg-accent rounded-full"
          href="#"
        >
          <img
            height="50px"
            src="/images/icon-line.png"
            width="50px"
          />
          <span
            className="text-white font-bold"
          >
            友達に追加して質問してみる
          </span>
        </a>
      </div>
    </div>
  </div>
</section>
`;
